modules/mariadb/src/main/java/org/testcontainers/containers/MariaDBR2DBCDatabaseContainerProvider.java

```diff
@@ -0,0 +1,29 @@
+package org.testcontainers.containers;
+
+import com.google.auto.service.AutoService;
+import io.r2dbc.spi.ConnectionFactoryOptions;
+import org.mariadb.r2dbc.MariadbConnectionFactoryProvider;
+import org.testcontainers.r2dbc.R2DBCDatabaseContainer;
+import org.testcontainers.r2dbc.R2DBCDatabaseContainerProvider;
+
+@AutoService(R2DBCDatabaseContainerProvider.class)
```

Comment:
Woah, TIL about `@AutoService`. I like it.

For the sake of consistency, we should probably propagate this out to everywhere else where we've currently hand-written a service descriptor file. This could be a subsequent PR, though.